source/common/http/path_utility.cc

```diff
@@ -1,43 +1,48 @@
 #include "common/http/path_utility.h"
 
-#include "common/chromium_url/url_canon.h"
-#include "common/chromium_url/url_canon_stdstring.h"
 #include "common/common/logger.h"
+#include "common/http/deprecated_path_canonicalizer.h"
 
 #include "absl/strings/str_join.h"
 #include "absl/strings/str_split.h"
 #include "absl/types/optional.h"
+#include "url/url_canon.h"
+#include "url/url_canon_stdstring.h"
 
 namespace Envoy {
 namespace Http {
 
 namespace {
-absl::optional<std::string> canonicalizePath(absl::string_view original_path) {
-  std::string canonical_path;
-  chromium_url::Component in_component(0, original_path.size());
-  chromium_url::Component out_component;
-  chromium_url::StdStringCanonOutput output(&canonical_path);
-  if (!chromium_url::CanonicalizePath(original_path.data(), in_component, &output,
-                                      &out_component)) {
-    return absl::nullopt;
-  } else {
-    output.Complete();
-    return absl::make_optional(std::move(canonical_path));
+absl::optional<std::string> canonicalizePath(absl::string_view original_path,
+                                             Runtime::Loader* runtime) {
+  if (runtime && !runtime->snapshot().deprecatedFeatureEnabled(
+                     "envoy.deprecated_features.use_forked_chromium_url", /*default_value=*/true)) {
```

Comment:
Do we want to default true or false?

Reply:
I think for now, until the URL parser is replaced, we set this "true". @yanavlasov WDYT?

Reply:
If we make it `false` now we could get some test mileage, which could be useful for a change like this. This change it not really related to the URL parser, so I think `false` as default is ok.

Reply:
Done. Changed using reloadable flag.